lib/slax_web/websockets/disable.ex

```diff
@@ -0,0 +1,199 @@
+defmodule SlaxWeb.Disable do
+  @moduledoc """
+  A module that handles Slack websocket payloads for disabling Slax interactions and
+  builds modal views for slack https://api.slack.com/reference/surfaces/views
+  """
+
+  alias Slax.Slack
+  alias Slax.Channels
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "shortcut",
+        "callback_id" => "disable_slax"
+      }) do
+    view = build_disable_view(%{trigger_id: trigger_id})
+
+    Slack.open_modal(%{trigger_id: trigger_id, view: view})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "shortcut",
+        "callback_id" => "enable_slax"
+      }) do
+    view = build_enable_view()
+
+    Slack.open_modal(%{trigger_id: trigger_id, view: view})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => _trigger_id,
+        "type" => "view_submission",
+        "view" => %{
+          "callback_id" => "disable_view",
+          "state" => %{
+            "values" => values
+          }
+        }
+      }) do
+    with %{
+           channel_id: channel_id,
+           name: name
+         } <-
+           parse_state_values(Map.values(values)) do
+      Channels.create_or_update_channel(channel_id, %{
+        name: name,
+        disabled: true
+      })
+
+      :ok
+    end
+  end
+
+  def handle_payload(%{
+        "trigger_id" => _trigger_id,
+        "type" => "view_submission",
+        "view" => %{
+          "callback_id" => "enable_view",
+          "state" => %{
+            "values" => values
+          }
+        }
+      }) do
+    case parse_state_values(Map.values(values)) do
+      %{
+        channel_id: "invalid",
+        name: _
+      } ->
+        :invalid
+
+      %{
+        channel_id: channel_id,
+        name: name
+      } ->
+        Channels.create_or_update_channel(channel_id, %{
+          name: name,
+          disabled: false
+        })
+
+        :ok
+    end
+  end
+
+  defp parse_state_values(values) do
+    with [
+           %{
+             "channel_select" => %{
+               "selected_option" => %{
+                 "text" => name,
+                 "value" => channel_id
+               },
+               "type" => "static_select"
+             }
+           }
+         ] <- values do
+      Map.new(%{
+        name: name["text"],
+        channel_id: channel_id
+      })
```

Comment:
Here's another unnecessary `with` (this one doesn't even need to be a case b/c there's only one option), and `Map.new()` doesn't need to be used if you're already feeding it a Map